### PR TITLE
[core] Universal getter for layer layout and light properties

### PR DESCRIPTION
**Pull request by pozdnyakov**
This PR enables fetching layer layout properties from the universal getter API `GetProperty()`, similar API is added to get light properties.
Having a similar method for getting both layout an paint layer properties decreases API surface and saves binary size.

### CODE REVIEW

File: expression-test/expression_test_parser.cpp [hunk 1]
```diff
@@ -439,33 +439,7 @@ optional<Value> toValue(const expression::Value& exprValue) {
             std::vector<Value> color { double(c.r), double(c.g), double(c.b), double(c.a) };
             return {Value{std::move(color)}};
```
Comment (alexshalamov) [inline expression-test/expression_test_parser.cpp]: color.toObject ?

File: src/mbgl/style/expression/formatted.cpp [hunk 2]
```diff
@@ -35,6 +35,35 @@ std::string Formatted::toString() const {
...
+    mapbox::base::ValueObject result;
+    mapbox::base::ValueArray sectionValues;
+    sectionValues.reserve(sections.size());
+    for (const auto& section : sections) {
+        mapbox::base::ValueObject serializedSection;
+        serializedSection.emplace("text", section.text);
+        if (section.fontScale) {
+            serializedSection.emplace("scale", *section.fontScale);
+        } else {
+            serializedSection.emplace("scale", NullValue());
```
Comment (alexshalamov) [inline src/mbgl/style/expression/formatted.cpp]: should it be mapbox::base::NullValue?

Comment (pozdnyakov) [reply src/mbgl/style/expression/formatted.cpp]: it's an alias to it

File: expression-test/expression_test_parser.cpp (hunk 1 above)
Comment (pozdnyakov) [reply expression-test/expression_test_parser.cpp]: in the follow up pr :-)  the representation there is different

**Comment by tmpsantos** [comment]
> No description provided

Better explanation here would be good. :-)

**Pull request merged**